### PR TITLE
feat(marketplace): add more english auction extensions

### PR DESCRIPTION
**Pull request by jnsdls**
closes: CNCT-658, CNCT-657, CNCT-662, CNCT-663, CNCT-907, CNCT-664, CNCT-665, CNCT-973

---

## PR-Codex overview
This PR introduces new functionalities related to English auctions in the marketplace extension.

### Detailed summary
- Added `nativeCurrency` field to Anvil chain definition
- Implemented `buyoutAuction` function in English auction
- Implemented `cancelAuction` function in English auction
- Implemented `getWinningBid` function in English auction
- Added `resolveCurrencyValue` function for currency conversion
- Implemented `executeSale` function in English auction
- Updated imports and exports for marketplace extension files

> The following files were skipped due to too many changes: `packages/thirdweb/src/extensions/marketplace/write/english-auction/bidInAuction.ts`, `packages/thirdweb/src/extensions/marketplace/marketplace.test.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by jnsdls** [comment]
4 more left to complete english auctions, will do those in the morning and add this

File: packages/thirdweb/src/extensions/marketplace/write/english-auction/bidInAuction.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+ * });
+ * ```
+ */
+export function bidInAuction(
+  options: BaseTransactionOptions<BidInAuctionParams>,
+) {
+  return generatedBidInAuction({
+    contract: options.contract,
+    asyncParams: async () => {
+      const [auction] = await Promise.all([
```
Comment (joaquim-verges) [inline packages/thirdweb/src/extensions/marketplace/write/english-auction/bidInAuction.ts:39]: nit: doesnt need the promise.all

File: packages/thirdweb/src/extensions/marketplace/write/english-auction/bidInAuction.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        getAuction({
+          contract: options.contract,
+          auctionId: options.auctionId,
+        }),
+      ]);
+
+      const resolvedBidAmountWei = await (async () => {
```
Comment (joaquim-verges) [inline packages/thirdweb/src/extensions/marketplace/write/english-auction/bidInAuction.ts:46]: remind the reason for adding a inline function here? does it trick the bundler to respect dynamic imports or something?

**Review by joaquim-verges — APPROVED** [review]

File: packages/thirdweb/src/extensions/marketplace/write/english-auction/bidInAuction.ts (hunk 2 above)
Comment (jnsdls) [reply packages/thirdweb/src/extensions/marketplace/write/english-auction/bidInAuction.ts:46]: no it simply nicely encapsulates the logic needed to resolve the actual bid amount in wei, the alternative is doing something along the lines of:
```
let bidAmount;
if("bidAmount" in options){
  ...
  bidAmount = ....
}
else {
...
}
```
which just makes my skin crawl from unreadability

File: packages/thirdweb/src/extensions/marketplace/write/english-auction/bidInAuction.ts (hunk 1 above)
Comment (jnsdls) [reply packages/thirdweb/src/extensions/marketplace/write/english-auction/bidInAuction.ts:39]: yep totally, will remove

**Pull request merged**